server/lib/models/workspace.ts

```diff
@@ -494,27 +499,52 @@ export default class Workspace extends Model<Workspace> {
     if (parentWorkspace === null) {
       return false;
     }
+
+    const rootWorkspace = await parentWorkspace.getRootWorkspace();
+    const tree = (await rootWorkspace.$get("tree")) as Tree;
+    const isMIBWithoutRestarts = tree.isMIBWithoutRestarts;
+
     const isParentRootWorkspace = !parentWorkspace.parentId;
-    const isParentOracleWorkspace =
-      parentWorkspace.isEligibleForHonestOracle ||
+    const isParentHonestOracleWorkspace =
+      parentWorkspace.isEligibleForHonestOracle;
+    const isParentMaliciousOracleWorkspace =
       parentWorkspace.isEligibleForMaliciousOracle;
 
-    if (isParentOracleWorkspace && !isParentRootWorkspace) {
+    // child of honest workspace never honest
+    if (isParentHonestOracleWorkspace) {
       return false;
     }
 
-    const rootWorkspace = await parentWorkspace.getRootWorkspace();
-
-    // get experiment id
-    const tree = (await rootWorkspace.$get("tree")) as Tree;
     const experiments = (await tree.$get("experiments")) as Experiment[];
 
     if (experiments.length === 0) {
       return false;
     }
 
     const mostRecentExperiment = _.sortBy(experiments, e => -e.createdAt)[0];
-    return mostRecentExperiment.areNewWorkspacesOracleOnlyByDefault;
+
+    // if experiment is non-oracle, then of course not honest
+    if (!mostRecentExperiment.areNewWorkspacesOracleOnlyByDefault) {
+      return false;
+    }
+
+    // if parent is root, then honest
+    if (isParentRootWorkspace) {
+      return true;
+    }
+
+    // if parent is non-root malicious, and MIB w/o restarts, then honest
+    if (isParentMaliciousOracleWorkspace && isMIBWithoutRestarts) {
+      return true;
+    }
+
+    // If parent is non-expert, then honest
+    if (!isParentMaliciousOracleWorkspace && !isParentHonestOracleWorkspace) {
+      return true;
+    }
+
+    // Else not honest
+    return false;
```

Comment:
@brachbach Does this logic match up with your expectations for honest children?

Reply:
Yep, though I'll admit I hadn't thought it through before and I now see that MIB will make the subtree view even weirder :p